README.md

```diff
@@ -1,126 +1,198 @@
+| 	min_length        | int     | false      | min character length.|
+| 	max_length        | int     | false      | max character length.|
+| 	min               | int     | false      |  min integer length. |
+| 	max               | int     | false      |  max integer length. |
+| 	options           | []string | false     |  specify the options when the vriable type is `enum`, for example: options:<br> - "ClusterIP" <br> - "NodePort" <br> - "LoadBalancer"|
+| 	valid_chars       | string   | false     |  regular expression for input chars validation. |
+| 	invalid_chars     | string   | false     |  regular expression for invalid input chars validation.|
+| 	subquestions      | []subquestion | false|  add an array of subquestions.|
+| 	show_if           | string      | false  | show current variable if conditional variable is true, for example `show_if: "serviceType=Nodeport"` |
+| 	show\_subquestion_if |  string  | false     | show subquestions if is true or equal to one of the options. for example `show_subquestion_if: "true"`|
 
-3. You can directly add `https://github.com/rancher/partner-charts` into rancher catalog. In order to show all the charts you have to run `make CHART=${chart_name} prepare` and make sure there is `chart-original` folder on each chart folder if your chart relies on a upstream chart.
+**subquestions**: `subquestions[]` cannot contain `subquestions` or `show_subquestions_if` keys, but all other keys in the above table are supported. 
 
-### Makefile
+### 6. Test your changes
 
-`make bootstrap`: 
+#### Generate modified chart
 
-Download binaries that are needed for ci scripts.
+To test your modified chart you will first need to generate it. The result is a tarball containing your upstream chart 
+with patch changes applied, and overlay files added on top of it. It will be located in the `assets` directory and as mentioned above, its name will be in the form `{CHART_NAME}-{VERSION}{packageVersion}.tgz`. 
 
-`make prepare`: 
+This will validate and generate a tarball of your modified chart
 
-Prepare the chart for modification. This will apply the upstream chart with the current patch. Use `CHART=${NAME}` for specific chart.
+```
+make validate CHART={CHART_NAME}
+```
 
-`make charts`: 
+Alternatively, this will skip validation and just generate the tarball of your modified chart (Validating it before submitting a pull request is recommended)
 
-Generate tarball for each charts. Use `CHART=${NAME}` for specific chart.
+```
+make charts CHART={CHART_NAME}
+```
 
-`make patch`: 
+#### Test modified chart
 
-Compare the current chart with upstream and generate patch file. Use `CHART=${NAME}` for specific chart. 
+An easy way to test your chart is by hosting the `assets` directory (Make sure you generate the tarball first) using 
+`python -m SimpleHTTPServer`, expose it using `ngrok`, and use the `ngrok` URL to add the chart as a custom catalog in Rancher.
```

Comment:
Just run `make charts`, push the changes to your fork, and add the fork + branch to the Repositories under the Dashboard UI and that should work. No need to do python / ngrok 😄 .

See above comment for how this can be rephrased.